### PR TITLE
filter only non deleted files

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Entity/Fileset.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function __construct()
     }
 
     /**
-     * Getter for files.
+     * Getter for all files.
      *
      * @return Collection
      */
```
Comment (mickel1138) [inline src/Entity/Fileset.php:37]: Maybe call this function getAllFiles?

File: tests/unit/Entity/FilesetTest.php [hunk 2]
```diff
@@ -41,8 +41,13 @@ class FilesetTest extends TestCase
     protected function setUp()
     {
         $this->fileset = new Fileset;
-        $this->mockFile = \Mockery::mock(File::class, array('setFileset' => null));
-        $this->defaultFile = $this->fileset->getFiles()->first();
+        $this->mockFile = \Mockery::mock(
+            File::class,
+            array(
+                'setFileset' => null,
+                'getStatus' => 'new'
```
Comment (mickel1138) [inline tests/unit/Entity/FilesetTest.php:48]: Use constant instead: `File::FILE_NEW`

File: tests/unit/Entity/FilesetTest.php [hunk 3]
```diff
@@ -52,17 +57,17 @@ protected function setUp()
      */
     public function testAddFileToFileset()
```
Comment (mickel1138) [inline tests/unit/Entity/FilesetTest.php:58]: Do we want test for all the other functions? If you look at my PR, maybe we can merge some of the functionality?

Comment (praneethpr) [reply tests/unit/Entity/FilesetTest.php:58]: I added those tests for the other getters

**Review by mickel1138 — APPROVED** [review]
Looks good.

**Pull request merged**